### PR TITLE
[SPARK-16870][docs]Summary:add "spark.sql.broadcastTimeout" into docs/sql-programming-gu…

### PR DESCRIPTION
**Pull request by biglobster**
## What changes were proposed in this pull request?

default value for spark.sql.broadcastTimeout is 300s. and this property do not show in any docs of spark. so add "spark.sql.broadcastTimeout" into docs/sql-programming-guide.md to help people to how to fix this timeout error when it happenned
## How was this patch tested?

not need

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

…ide.md

JIRA_ID:SPARK-16870
Description:default value for spark.sql.broadcastTimeout is 300s. and this property do not show in any docs of spark. so add "spark.sql.broadcastTimeout" into docs/sql-programming-guide.md to help people to how to fix this timeout error when it happenned
Test:done

### CODE REVIEW

**Comment by srowen** [comment]
Many properties are not documented on purpose, because they're semi-private. What's the argument that this was intended to be public and set by callers?

**Comment by biglobster** [comment]
I run a large number jobs with spark-sql at the same time. and meet the error that print  timeout (some job  contains the broadcast-join operator) : 
java.util.concurrent.TimeoutException: Futures timed out after [300 seconds]
        at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:219)
        at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
        at scala.concurrent.Await$$anonfun$result$1.apply(package.scala:107)
        at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
        at scala.concurrent.Await$.result(package.scala:107)
        at org.apache.spark.sql.execution.joins.BroadcastHashOuterJoin.doExecute(BroadcastHashOuterJoin.scala:113)
so if I reset value bigger then 300s, error do not show again

**Comment by srowen** [comment]
Any objection to documenting it @liancheng ?

File: docs/sql-programming-guide.md [hunk 1]
```diff
@@ -790,6 +790,15 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
     </p>
   </td>
 </tr>
+<tr>
```
Comment (srowen) [inline docs/sql-programming-guide.md]: PS this does not look like the right place for this option. This section covers Parquet. There's a later "other options" section covering broadcast join. Move it there.

Comment (biglobster) [inline docs/sql-programming-guide.md]: done. thx :) for you suggestion

**Comment by srowen** [comment]
Jenkins test this please

**Comment by srowen** [comment]
Merged to master/2.0

**Comment by liancheng** [comment]
@srowen Sorry for the late reply, this LGTM.

**Pull request closed without merging**